libc-bottom-half/sources/posix.c

```diff
@@ -214,13 +251,16 @@ int symlink(const char *target, const char *linkpath) {
     return symlinkat(target, dirfd, relative_path);
 }
 
+static char *relative_buf2 = NULL;
+static size_t relative_buf_len2 = NULL;
```

Comment:
It would tidy up the callsites that use these to have a function that holds these as function-local statics, similar to `find_relpath`.